engines/e_dasync.c

```diff
@@ -647,6 +650,8 @@ static int dasync_cipher_init_key_helper(EVP_CIPHER_CTX *ctx,
     struct dasync_pipeline_ctx *pipe_ctx =
         (struct dasync_pipeline_ctx *)EVP_CIPHER_CTX_get_cipher_data(ctx);
 
+    if (cipher == NULL)
+        return -1;
```

Comment:
Should this return `0` to stay consistent in this function?

Reply:
I wasn't sure how to handle this either, but looking at the code, the return 0 happens after an error is raised for a memory allocation problem - not something to do with this method, whereas in this case one of the arguments is essentially wrong. I'll have a better look when I get to the keyboard